app/main.py

```diff
@@ -1,213 +1,64 @@
-        "name": "Red Knight",
-        "power": 40,
-        "hp": 70,
-        "armour": [
-            {
-                "part": "breastplate",
-                "protection": 25,
-            }
-        ],
-        "weapon": {
-            "name": "Sword",
-            "power": 45
-        },
-        "potion": {
-            "name": "Blessing",
-            "effect": {
-                "hp": +10,
-                "power": +5,
-            }
-        }
-    }
-}
-
-
-def battle(knightsConfig):
-    # BATTLE PREPARATIONS:
-
-    # lancelot
-    lancelot = knightsConfig["lancelot"]
-
-    # apply armour
-    lancelot["protection"] = 0
-    for a in lancelot["armour"]:
-        lancelot["protection"] += a["protection"]
-
-    # apply weapon
-    lancelot["power"] += lancelot["weapon"]["power"]
-
-    # apply potion if exist
-    if lancelot["potion"] is not None:
-        if "power" in lancelot["potion"]["effect"]:
-            lancelot["power"] += lancelot["potion"]["effect"]["power"]
-
-        if "protection" in lancelot["potion"]["effect"]:
-            lancelot["protection"] += lancelot["potion"]["effect"]["protection"]
-
-        if "hp" in lancelot["potion"]["effect"]:
-            lancelot["hp"] += lancelot["potion"]["effect"]["hp"]
-
-    # arthur
-    arthur = knightsConfig["arthur"]
+from app.startStats import KNIGHTS
 
-    # apply armour
-    arthur["protection"] = 0
-    for a in arthur["armour"]:
-        arthur["protection"] += a["protection"]
 
-    # apply weapon
-    arthur["power"] += arthur["weapon"]["power"]
+class King:
+    def __init__(self, name, power, hp, protection=0):
+        self.name = name
+        self.power = power
+        self.hp = hp
+        self.protection = protection
 
-    # apply potion if exist
-    if arthur["potion"] is not None:
-        if "power" in arthur["potion"]["effect"]:
-            arthur["power"] += arthur["potion"]["effect"]["power"]
+    def new_stat(self, name):
+        for a in name["armour"]:
+            self.protection += a["protection"]
 
-        if "protection" in arthur["potion"]["effect"]:
-            arthur["protection"] += arthur["potion"]["effect"]["protection"]
+        # apply weapon
+        self.power += name["weapon"]["power"]
 
-        if "hp" in arthur["potion"]["effect"]:
-            arthur["hp"] += arthur["potion"]["effect"]["hp"]
+        # apply potion if exist
+        if name["potion"] is not None:
+            if "power" in name["potion"]["effect"]:
+                self.power += name["potion"]["effect"]["power"]
 
-    # mordred
-    mordred = knightsConfig["mordred"]
+            if "protection" in name["potion"]["effect"]:
+                self.protection += name["potion"]["effect"]["protection"]
 
-    # apply armour
-    mordred["protection"] = 0
-    for a in mordred["armour"]:
-        mordred["protection"] += a["protection"]
+            if "hp" in name["potion"]["effect"]:
+                self.hp += name["potion"]["effect"]["hp"]
```

Comment:
These code duplications can be also fixed, using `stats = ("protection", "power", "hp")` and looping on `stats`

Reply:
I don't understand how to do it. there is only one idea, but it does not work)

Reply:
Try to use this:
`setattr(self, stat, getattr(self, stat) + self.potion["effect"][stat])`